### PR TITLE
Add CodeCoverage (CodeCov.IO) support

### PR DESCRIPTION
**Pull request by 304NotModified**
CodeCov.IO is a free (for FOSS) Code coverage platform in the cloud. You can sign up with your GitHub account and add in the future badges and webhooks. 

This PR will submit the code coverage to CodeCov.IO

### CODE REVIEW

**Comment by 304NotModified** [comment]
Coverage on my branch:

image

https://codecov.io/gh/304NotModified/messagetemplates-csharp/tree/appveyor/src/MessageTemplates

File: Build.ps1 [hunk 1]
```diff
@@ -17,13 +17,6 @@ if($LASTEXITCODE -ne 0) { exit 1 }
 
 Pop-Location
 
-Push-Location test\MessageTemplates.Tests
-
-& dotnet test -c Release
```
Comment (adamchester) [inline Build.ps1:22]: Would it be possible to keep this so that local build & test workflow remains unchanged?

Comment (304NotModified) [inline Build.ps1:22]: Sure, but what's the benefit on running the same unit tests twice?

Comment (adamchester) [inline Build.ps1:22]: I wouldn't want them to run twice of course, but I do need them to run locally without waiting for AppVeyor, right?

Comment (304NotModified) [inline Build.ps1:22]: Good point. I have removed the commit :)

**Comment by 304NotModified** [comment]
I think this could be merged?

**Comment by adamchester** [comment]
I was thinking, if it's not too much trouble, we could avoid running twice on appveyor by disabling the normal tests (only if running under appveyor)?

**Comment by 304NotModified** [comment]
Sure, but it saves you 1 minute on build time at top?

**Pull request merged**